File: lib/data/models/player.dart
import 'package:json_annotation/json_annotation.dart';

part 'player.g.dart';

@JsonSerializable(explicitToJson: true)
class Player {
  final String id;
  final String name;
  final int seat;
  @JsonKey(defaultValue: false)
  final bool ready;
  @JsonKey(defaultValue: 0)
  final int cardCount;
  @JsonKey(defaultValue: false)
  final bool isLandlord;
  @JsonKey(defaultValue: -1)
  final int bidValue; // 玩家叫分

  Player({
    required this.id,
    required this.name,
    required this.seat,
    required this.ready,
    this.cardCount = 0,
    this.isLandlord = false,
    this.bidValue = -1,
  });

  factory Player.fromJson(Map<String, dynamic> json) => _$PlayerFromJson(json);
  Map<String, dynamic> toJson() => _$PlayerToJson(this);
}

File: server_3/src/models/Player.ts
import { Poker } from "./Poker";


export class Player {
    id: string; //同时也是socketid
    name: string;
    seat: number = 0;
    cardCount: number = 0;
    ready: boolean = false;
    isLandlord = false;
    bidValue = -1;

    constructor(socketId: string, name: string, seat: number) {
        this.id = socketId;
        this.name = name || `玩家_${Math.random().toString(36).substr(2, 4)}`;
        this.seat = seat;
    }
}